clang/lib/DPCT/APINamesTemplateType.inc

```diff
@@ -58,94 +58,169 @@ TYPE_REWRITE_ENTRY("cub::TransformInputIterator",
-                       TYPE_CONDITIONAL_FACTORY(
-                           CheckTemplateArgCount(4),
-                           TYPE_FACTORY(STR(MapNames::getDpctNamespace() +
-                                            "group::group_load"),
-                                        TEMPLATE_ARG(0), TEMPLATE_ARG(2),
-                                        TEMPLATE_ARG(3)),
-                           TYPE_FACTORY(STR(MapNames::getDpctNamespace() +
-                                            "group::group_load"),
-                                        TEMPLATE_ARG(0), TEMPLATE_ARG(2)))))
-TYPE_REWRITE_ENTRY("cub::BlockStore",
+    "cub::ArgMax",
+    HEADER_INSERTION_FACTORY(
+        HeaderType::HT_DPCT_DPL_Utils,
+        TYPE_CONDITIONAL_FACTORY(
+            UseSYCLCompat(),
+            WARNING_FACTORY(Diagnostics::UNSUPPORT_SYCLCOMPAT, TYPESTR),
+            TYPE_FACTORY(STR(MapNames::getLibraryHelperNamespace() + "argmax")))))
+
+// cub::ArgMin
+TYPE_REWRITE_ENTRY(
+    "cub::ArgMin",
+    HEADER_INSERTION_FACTORY(
+        HeaderType::HT_DPCT_DPL_Utils,
+        TYPE_CONDITIONAL_FACTORY(
+            UseSYCLCompat(),
+            WARNING_FACTORY(Diagnostics::UNSUPPORT_SYCLCOMPAT, TYPESTR),
+            TYPE_FACTORY(STR(MapNames::getLibraryHelperNamespace() + "argmin")))))
+
+// cub::BlockRadixSort
+TYPE_REWRITE_ENTRY("cub::BlockRadixSort",
                    HEADER_INSERTION_FACTORY(
                        HeaderType::HT_DPCT_GROUP_Utils,
                        TYPE_CONDITIONAL_FACTORY(
-                           CheckTemplateArgCount(4),
-                           TYPE_FACTORY(STR(MapNames::getDpctNamespace() +
-                                            "group::group_store"),
-                                        TEMPLATE_ARG(0), TEMPLATE_ARG(2),
-                                        TEMPLATE_ARG(3)),
-                           TYPE_FACTORY(STR(MapNames::getDpctNamespace() +
-                                            "group::group_store"),
+                           UseSYCLCompat(),
+                           WARNING_FACTORY(Diagnostics::UNSUPPORT_SYCLCOMPAT,
+                                           TYPESTR),
+                           TYPE_FACTORY(STR(MapNames::getLibraryHelperNamespace() +
+                                            "group::group_radix_sort"),
                                         TEMPLATE_ARG(0), TEMPLATE_ARG(2)))))
+
+// cub::BlockExchange
+TYPE_REWRITE_ENTRY(
+    "cub::BlockExchange",
+    HEADER_INSERTION_FACTORY(
+        HeaderType::HT_DPCT_GROUP_Utils,
+        TYPE_CONDITIONAL_FACTORY(
+            UseSYCLCompat(),
+            WARNING_FACTORY(Diagnostics::UNSUPPORT_SYCLCOMPAT, TYPESTR),
+            TYPE_FACTORY(STR(MapNames::getLibraryHelperNamespace() + "group::exchange"),
+                         TEMPLATE_ARG(0), TEMPLATE_ARG(2)))))
+// cub::BlockLoad
+TYPE_REWRITE_ENTRY(
+    "cub::BlockLoad",
+    HEADER_INSERTION_FACTORY(
```

Comment:
Don't need to insert help headers when using syclcompat.

Reply:
Done